src/lib/decorators/ack/AckDecoratorExtension.ts

```diff
@@ -0,0 +1,30 @@
+import { Expose, Type } from 'class-transformer';
+import { ValidateNested } from 'class-validator';
+
+import { BaseMessageConstructor } from '../../agent/BaseMessage';
+import { AckDecorator } from './AckDecorator';
+
+export function AckDecorated<T extends BaseMessageConstructor>(Base: T) {
+  class AckDecoratorExtension extends Base {
+    @Expose({ name: '~please_ack' })
+    @Type(() => AckDecorator)
+    @ValidateNested()
+    public pleaseAck?: AckDecorator;
+
+    public setPleaseAck(pleaseAck: Record<string, unknown>) {
+      this.pleaseAck = new AckDecorator({
+        pleaseAck,
+      });
+    }
+
+    public getPleaseAck(): AckDecorator | undefined {
+      return this.pleaseAck;
+    }
+
+    public requiresAck(): boolean {
+      return !!this.pleaseAck;
```

Comment:
Hmm. I see your point but what about the `null` check? Do we want to just rely on TS? What if `null` will be allowed option in the future versions of the protocol? 🤔 

I'm thinking if we should introduce some util function like (inspired by lodash):
```javascript
function isNil(value) {
  return value == null;
}
```

Reply:
Interesting point. Theoretically it shouldn't be null. It can either be `undefined` or an (possibly empty) object. It's not specified explicitly in the RFCS, [but null is not the same as <missing>](https://github.com/hyperledger/aries-rfcs/pull/527#issuecomment-674124087) and therefore I don't think we have to (and should account) for it except for maybe using class-validator to disallow `null`. But if `null` would be an allowed option in the future I think the `isNil` function would be of good use.